alacritty/src/display.rs

```diff
@@ -394,68 +395,31 @@ impl Display {
             api.clear(background_color);
         });
 
-        let mut lines = RenderLines::new();
         let mut urls = Urls::new();
+        let mut rects: Vec<RenderRect> = Vec::new();
 
         // Draw grid
         {
             let _sampler = self.meter.sampler();
 
             self.renderer.with_api(&config, &size_info, |mut api| {
                 // Iterate over all non-empty cells in the grid
-                for cell in grid_cells {
-                    // Update URL underlines
-                    urls.update(size_info.cols().0, cell);
-
+                for text_run in grid_text_runs {
                     // Update underline/strikeout
-                    lines.update(cell);
+                    if text_run.flags.contains(Flags::UNDERLINE) {
+                        let underline_metrics = (metrics.descent, metrics.underline_position, metrics.underline_thickness);
+                        rects.push(RenderRect::from_text_run(&text_run, underline_metrics, &size_info));
+                    }
+                    if text_run.flags.contains(Flags::STRIKEOUT) {
+                        let strikeout_metrics = (metrics.descent, metrics.strikeout_position, metrics.strikeout_thickness);
+                        rects.push(RenderRect::from_text_run(&text_run, strikeout_metrics, &size_info));
+                    }
 
                     // Draw the cell
-                    api.render_cell(cell, glyph_cache);
+                    api.render_text_run(text_run, glyph_cache);
                 }
             });
         }
-
-        let mut rects = lines.rects(&metrics, &size_info);
-
-        // Update visible URLs
-        self.urls = urls;
-        if let Some(url) = self.urls.highlighted(config, mouse, mods, mouse_mode, selection) {
-            rects.append(&mut url.rects(&metrics, &size_info));
-
-            self.window.set_mouse_cursor(CursorIcon::Hand);
-
-            self.highlighted_url = Some(url);
-        } else if self.highlighted_url.is_some() {
-            self.highlighted_url = None;
-
-            if mouse_mode {
-                self.window.set_mouse_cursor(CursorIcon::Default);
-            } else {
-                self.window.set_mouse_cursor(CursorIcon::Text);
-            }
-        }
-
-        // Highlight URLs at the vi mode cursor position
-        if let Some(vi_mode_cursor) = vi_mode_cursor {
-            if let Some(url) = self.urls.find_at(vi_mode_cursor.point) {
-                rects.append(&mut url.rects(&metrics, &size_info));
-            }
-        }
-
-        // Push visual bell after url/underline/strikeout rects
-        if visual_bell_intensity != 0. {
-            let visual_bell_rect = RenderRect::new(
-                0.,
-                0.,
-                size_info.width,
-                size_info.height,
-                config.visual_bell.color,
-                visual_bell_intensity as f32,
-            );
-            rects.push(visual_bell_rect);
-        }
-
```

Comment:
This code wasn't removed in the last PR: https://github.com/alacritty/alacritty/pull/2677/files#diff-ded871246e5124657885a75595b713d7L403

I guess it should be kept?